client/services/FeedbackGeneratorServiceSpec.js

```diff
@@ -81,10 +83,61 @@ describe('FeedbackGeneratorService', function() {
             c: 'j'
           })
         ).toEqual('{"a": 3, "b": 5, "c": "j"}');
+        // Need to define and call blankFunction for 100% karma coverage
+        var blankFunction = function() {
+          return null;
+        };
+        blankFunction();
+        expect(
+          FeedbackGeneratorService._jsToHumanReadable(blankFunction)
+        ).toEqual('[UNKNOWN OBJECT]');
       }
     );
   });
 
+  describe('_getCorrectnessTestFeedback', function() {
+    it('should return output of user function with issues', function() {
+      var correctnessTest = CorrectnessTestObjectFactory.create({
+        input: 'cat',
+        allowedOutputs: ['a']
+      });
+
+      var paragraphs = FeedbackGeneratorService
+        ._getCorrectnessTestFeedback(correctnessTest, "output").getParagraphs();
+      expect(paragraphs.length).toEqual(5);
+      expect(paragraphs[0].isTextParagraph()).toBe(true);
+      expect(paragraphs[0].getContent()).toBe(
+        "Your code produced the following result:");
+      expect(paragraphs[1].isCodeParagraph()).toBe(true);
+      expect(paragraphs[1].getContent()).toBe(
+        "Input: \"cat\"\nOutput: \"output\"");
+      expect(paragraphs[2].isTextParagraph()).toBe(true);
+      expect(paragraphs[2].getContent()).toBe(
+        "However, the expected output is:");
+      expect(paragraphs[3].isCodeParagraph()).toBe(true);
+      expect(paragraphs[3].getContent()).toBe("\"a\"");
+      expect(paragraphs[4].isTextParagraph()).toBe(true);
+      expect(paragraphs[4].getContent()).toBe(
+        "Could you fix this?");
+    });
+  });
+
+  describe('_getPerformanceTestFeedback', function() {
+    it('should return output of user function performance slow', function() {
```

Comment:
should return feedback if user's function is running significantly slower than expected